src/hooks/useTroubleshootingPanel.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { Action, ExtensionHook, useModal } from '@openshift-console/dynamic-plugin-sdk';
+import { InfrastructureIcon } from '@patternfly/react-icons';
+import Popover from '../components/Popover';
+import { useBoolean } from './useBoolean';
+import { useDispatch } from 'react-redux';
+import { useKorrel8r } from './useKorrel8r';
+import { openTP } from '../redux-actions';
+import { useURLState } from './useURLState';
+
+const useTroubleshootingPanel: ExtensionHook<Array<Action>> = () => {
+  const { isKorrel8rReachable } = useKorrel8r();
+  const { korrel8rQueryFromURL } = useURLState();
+  const launchModal = useModal();
+  const dispatch = useDispatch();
+  const [isLaunched, , setLaunched] = useBoolean(false);
+  const open = React.useCallback(() => {
+    dispatch(openTP());
+  }, [dispatch]);
+
+  const [actions, setActions] = React.useState<Array<Action>>([
+    {
```

Comment:
IIUC this could be extracted into a function that receives the queryString, by default `''` and `korrel8rQueryFromURL` tracked by the useEffect

Reply:
Move this out into a separate function